ucrop/src/main/java/com/yalantis/ucrop/task/BitmapCropTask.java

```diff
@@ -149,19 +151,36 @@ private boolean crop() throws IOException {
         Log.i(TAG, "Should crop: " + shouldCrop);
 
         if (shouldCrop) {
-            ExifInterface originalExif = new ExifInterface(mImageInputPath);
             saveImage(Bitmap.createBitmap(mViewBitmap, cropOffsetX, cropOffsetY, mCroppedImageWidth, mCroppedImageHeight));
             if (mCompressFormat.equals(Bitmap.CompressFormat.JPEG)) {
-                ImageHeaderParser.copyExif(originalExif, mCroppedImageWidth, mCroppedImageHeight, mImageOutputPath);
+                if (mImageInputUri != null && "content".equals(mImageInputUri.getScheme()) && "content".equals(mImageOutputUri.getScheme())) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        ImageHeaderParser.copyExif(mContext.get(), mCroppedImageWidth, mCroppedImageHeight, mImageInputUri, mImageOutputUri);
+                    } else {
+                        Log.e(TAG, "It is not possible to write exif info into file represented by \"content\" Uri if Android < LOLLIPOP");
+                    }
+                } else if (mImageInputUri != null && "content".equals(mImageInputUri.getScheme())) {
+                    ImageHeaderParser.copyExif(mContext.get(), mCroppedImageWidth, mCroppedImageHeight, mImageInputUri, mImageOutputPath);
+                } else if (mImageOutputUri != null && "content".equals(mImageOutputUri.getScheme())) {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                        ExifInterface originalExif = new ExifInterface(mImageInputPath);
+                        ImageHeaderParser.copyExif(mContext.get(), originalExif, mCroppedImageWidth, mCroppedImageHeight, mImageOutputUri);
+                    } else {
+                        Log.e(TAG, "It is not possible to write exif info into file represented by \"content\" Uri if Android < LOLLIPOP");
+                    }
+                } else {
+                    ExifInterface originalExif = new ExifInterface(mImageInputPath);
+                    ImageHeaderParser.copyExif(originalExif, mCroppedImageWidth, mCroppedImageHeight, mImageOutputPath);
```

Comment:
Since the previous `copyExif` call is wrapped into build version check, we need it here too

Reply:
No we don't. The reason is the same as explained earlier. Here we use `mImageOutputPath`. We just need build versions checks if we are using `mImageOutputUri` as the destination of an exif copy.

Reply:
Got it.
I'd like to move this chain of `if else` statements to ImageHeaderParser. We could leave it as is and I'll do it in another PR